src/posix.c

```diff
@@ -62,8 +62,11 @@ int p_getaddrinfo(
 	ai = ainfo;
 
 	for (p = 1; ainfo->ai_hostent->h_addr_list[p] != NULL; p++) {
-		ai->ai_next = malloc(sizeof(struct addrinfo));
-		memcpy(&ai->ai_next, ainfo, sizeof(struct addrinfo));
+		if (!(ai->ai_next = malloc(sizeof(struct addrinfo)))) {
```

Comment:
Am 2015-12-31 16:19, schrieb Carlos Martín Nieto:
> In src/posix.c [1]:
> 
>> @@ -62,8 +62,11 @@ int p_getaddrinfo(
>>      ai = ainfo;
>> 
>>      for (p = 1; ainfo->ai_hostent->h_addr_list[p] != NULL; p++) {
>> -        ai->ai_next = malloc(sizeof(struct addrinfo));
>> -        memcpy(&ai->ai_next, ainfo, sizeof(struct addrinfo));
>> +        if (!(ai->ai_next = malloc(sizeof(struct addrinfo)))) {
> 
> We should be using git__malloc() here, or even git__calloc(). And yes,
> we should just bail out with the macro.

Note however that this is a replacement function for getaddrinfo() that 
also doesn't follow libgit2 semantics. For instance, GITERR_CHECK_ALLOC 
will not clean up properly while the standard getaddrinfo() is supposed 
to cleanup properly in a failure case. So should this really being used? 
In posix.c there is quite a mixture of styles so it is not clear what 
style should be really used. But if you say git__malloc() and 
GITERR_CHECK_ALLOC then I will adjust my changes.